build-python.sh

```diff
@@ -362,9 +362,10 @@ if test "${INSTALL}" = true; then
     # ref for use of '--find-links': https://stackoverflow.com/a/52481267/3986677
     pip install \
         ${PIP_INSTALL_ARGS} \
-        --ignore-installed \
```

Comment:
`--ignore-installed` was added in #6526 for the following reason:
> It also proposes remove the `pip uninstall lightgbm` in that script with passing `--ignore-installed` to `pip install`... that way the locally-built version overwrites the already-installed one every time, without generating a log message like *"WARNING: Skipping lightgbm as it is not installed."*

But it seems that the right flag is [`--force-reinstall`](https://pip.pypa.io/en/stable/cli/pip_install/#cmdoption-force-reinstall) for that.
`--ignore-installed` may mix old and new lightgbm installations which can result in corrupted installation. Refer to the detailed explanation here: https://stackoverflow.com/a/51916623.

Reply:
You're right, `--force-reinstall` is preferable now that we are also using `--no-deps` here (#6256).

I'd previously not wanted to use it because it applies to all dependencies, not just to the package being installed, and that led to an existing `numpy` or `scipy` being force-reinstalled too.